cmd/juju/application/deploy.go

```diff
@@ -794,9 +794,54 @@ func (c *DeployCommand) maybeReadLocalBundle() (deployFn, error) {
 	}, nil
 }
 
-func (c *DeployCommand) maybeReadLocalCharm() (deployFn, error) {
+func isNotExistsError(err error) bool {
+	if os.IsNotExist(err) {
+		return true
+	}
+	// On Windows, we get a path error due to a GetFileAttributesEx syscall.
+	// To avoid being too proscriptive, we'll simply check for the error
+	// type and not any content.
+	if _, ok := err.(*os.PathError); ok {
+		return true
+	}
+	return false
+}
+
+func (c *DeployCommand) maybeReadLocalCharm(apiRoot DeployAPI) (deployFn, error) {
+	//NOTE: Here we select the series using the algorithm defined by the
+	//series selector. This serves to override the algorithm found in
+	//`charmrepo.NewCharmAtPath` which is outdated (but must still be called
+	//since the code is coupled with path interpretation logic which needs
+	//to be factored out).
+
```

Comment:
IMHO, the `seriesSelector` should be moved into the charmrepo package as part of a refactoring of the package. For example the `charmrepo.NewCharmAtPathForceSeries` both serves to determine if the charm reference can be interpreted as a local charm path and also determines its series. The existence of the `seriesSelector` illustrates the need to have these concerns separate.